### PR TITLE
Help: Implement a loading placeholder the contact form

### PR DESCRIPTION
**Pull request by dllh**
When loading the contact form, we have to fetch information about the availability of chat. The loading screen isn't very useful right now, as described in #515. Here's what it looks like:

loading

A user who sees this screen might reasonably think that something's broken. To repro:

1. Go to http://calypso.localhost:3000/help/contact
2. Observe the blinking placeholder

This PR attempts to improve the UX by adding some placeholder form controls and styles to more closely mirror similar loading experiences elsewhere in the product. Here it is in action:

chat-loading

Cc @folletto since you had kindly offered feedback on #515. Cc @omarjackman.

### CODE REVIEW

**Comment by dllh** [comment]
Will need some styling love in Firefox:

screen shot 2015-12-16 at 4 26 36 pm

File: client/me/help/help-contact/index.jsx [hunk 1]
```diff
@@ -260,7 +263,13 @@ module.exports = React.createClass( {
 		}
 
 		if ( ! ( olark.isOlarkReady && sitesInitialized ) ) {
-			return <div className="help-contact__placeholder" />;
+			return (
```
Comment (omarjackman) [inline client/me/help/help-contact/index.jsx]: How about you use the actual contact form and just add the placeholder styles.
do something like:

```js
return (
    <div className="help-contact__placeholder">
        <HelpContactForm disabled={true} showSubjectField={ true } />
    </div>
);
```

File: client/me/help/help-contact/index.jsx [hunk 2]
```diff
@@ -260,7 +260,14 @@ module.exports = React.createClass( {
...
+				<div className="help-contact__placeholder">
+					<HelpContactForm disabled={true}
+						showSubjectField={ false }
+						onSubmit={ function() {} }
+						buttonLabel={ this.translate( 'Contact us' ) }/>
+				</div>
+			);
 		}
 
 		if ( olark.details.isConversing && olark.isOperatorAvailable ) {
```
Comment (dllh) [inline client/me/help/help-contact/index.jsx]: Ah, thanks. I've removed the subject field (it rendered oddly with the placeholder, and we won't have that for chats anyway). I'm not sure about my treatment of the `onSubmit` prop, which is required and must be a function. This feels a little hacky, and I'm not sure whether there's a better syntax to use here or whether instead I should change the validation (which doesn't feel like the best thing to do).

Comment (omarjackman) [inline client/me/help/help-contact/index.jsx]: I suggested this because I saw it done in `HelpSearch`. If it seems hacky then maybe it is.

Comment (omarjackman) [inline client/me/help/help-contact/index.jsx]: I've also seen people create a `PlaceHolder` sub component. The hacky way I guess just seems a bit to easy.

File: client/me/help/help-contact/index.jsx [hunk 3]
```diff
@@ -260,7 +260,14 @@ module.exports = React.createClass( {
 		}
 
 		if ( ! ( olark.isOlarkReady && sitesInitialized ) ) {
-			return <div className="help-contact__placeholder" />;
+			return (
+				<div className="help-contact__placeholder">
+					<HelpContactForm disabled={true}
+						showSubjectField={ false }
+						onSubmit={ function() {} }
```
Comment (omarjackman) [inline client/me/help/help-contact/index.jsx]: You can use lodash `noop` here if using an empty function makes this feel hacky to you.

Comment (dllh) [inline client/me/help/help-contact/index.jsx]: Ah, cool, I'll update with that when I'm back on later. (And to clarify, it was this particular thing that I meant felt hacky, not the general thing you had proposed, which seems pretty neat.)

**Comment by omarjackman** [comment]
@dllh I'd also style the submit button with a placeholder so that the user doesn't see the button text.

File: client/me/help/help-contact/index.jsx [hunk 4]
```diff
@@ -260,7 +260,14 @@ module.exports = React.createClass( {
 		}
 
 		if ( ! ( olark.isOlarkReady && sitesInitialized ) ) {
-			return <div className="help-contact__placeholder" />;
+			return (
+				<div className="help-contact__placeholder">
+					<HelpContactForm disabled={true}
+						showSubjectField={ false }
```
Comment (omarjackman) [inline client/me/help/help-contact/index.jsx]: Don't need the `showSubjectField` here since the default value is `false`. See [here](https://github.com/Automattic/wp-calypso/blob/update/olark-loading-screen/client/me/help/help-contact-form/index.jsx#L37) for all the defaults.

File: client/me/help/help-contact/style.scss [hunk 5]
```diff
@@ -1,5 +1,10 @@
...
 	height: 237px;
 	width: 100%;
-}
+	.form-textarea {
+		@include placeholder();
+	}
+	.form-label {
+		@include placeholder();
+	}
+}
```
Comment (omarjackman) [inline client/me/help/help-contact/style.scss]: Add line break at the end of the file.

File: client/me/help/help-contact/style.scss [hunk 6]
```diff
@@ -1,5 +1,10 @@
...
-	@include placeholder();
 	height: 237px;
 	width: 100%;
-}
+	.form-textarea {
+		@include placeholder();
+	}
+	.form-label {
+		@include placeholder();
+	}
```
Comment (omarjackman) [inline client/me/help/help-contact/style.scss]: These can probably be consolidated to:

```css
.form-textarea,
.form-label,
.form-button {
    @include placeholder();
}
```
Also don't forget to make the form button a placeholder.

File: client/me/help/help-contact/style.scss [hunk 7]
```diff
@@ -1,5 +1,10 @@
 .help-contact__placeholder {
-	@include placeholder();
 	height: 237px;
```
Comment (omarjackman) [inline client/me/help/help-contact/style.scss]: We no longer need a `height` here

File: client/me/help/help-contact/style.scss [hunk 8]
```diff
@@ -1,5 +1,10 @@
...
 	width: 100%;
```
Comment (omarjackman) [inline client/me/help/help-contact/style.scss]: We no longer need a `width` here

**Comment by folletto** [comment]
Note that the idea behind skeletons is that they are just shapes, with nothing characterizing them: not buttons, not styles, nor bordered windows. It's literally just a pulsating shape. 

If you don't feel the proposed design in #515 is good, let's find a different pattern of shapes, but still needs to be just rectangles. :)

**Comment by dllh** [comment]
@omarjackman I've fixed up all the issues you highlighted and only afterward saw @folletto's feedback. 

Davide, I wasn't trying to reject any design from #515. It's more that I thought you were showing some samples of similar things. I assumed we wanted the shapes to resemble their actual final forms somewhat faithfully and so attempted to reproduce approximations of the final forms. With Omar's suggestion of using the actual contact form component and just styling it, I thought we had a pretty faithful shadow representation.

In other words, I think the designs offered in #515 work well for what they're designed for but also thought they differed enough visually from what we finally display here that they weren't something I should just drop in. Do you think just using a couple of rectangles resembling what the placeholder component looks like (minus the button) would work here or do you think we need lots of extra rectangles?

**Comment by dllh** [comment]
@folletto would something like this do the trick or does it still try too closely to mimic the final view?

chat-loading4

**Comment by dikiaap** [comment]
wale-oke1

**Comment by folletto** [comment]
> Davide, I wasn't trying to reject any design from #515. It's more that I thought you were showing some samples of similar things. I assumed we wanted the shapes to resemble their actual final forms somewhat faithfully and so attempted to reproduce approximations of the final forms. 

The [first one](https://cloud.githubusercontent.com/assets/4389/11652285/c076c630-9d8e-11e5-85fe-c2aadde07303.png) was a design for the skeleton, which loosely matched all the three possible forms there, and in these all the three blocks with the text label above. :)

But it's ok! The above works too. :)

**Comment by dllh** [comment]
Thanks for clarifying, @folletto. I'm happy to go with the placeholder proposed by a designer. The current code now renders this:

chat-loading5

If that looks ok design-wise, then hopefully a quick glance at the code by @omarjackman will let us land this.

**Comment by blowery** [comment]
looks good! :ship:
